### ISSUE TITLE
LG webOS App won't find a TV

### ISSUE THREAD

**Issue by peaoo** [athombv/homey#204]
Hey, I can't connect my LG webOS Tv to Homey via the app.
If i try to pair them I get the popup: "There were no new devices found! You may close this window".
I've got the LG Connect settings enabled. 
My TV: LG 55UB850V-ZD
WebOS Version: 1.4.0-2309(afro-ashley)

And other user has the same problem:

TV Model: LG 47LB730V-ZD
WebOS Version: 1.4.0-2515 (afro-ashley)

**Comment by edtst** [comment]
Same here

**Comment by PandaNL** [comment]
Yep, same error.

**Comment by WeeJeWel** [comment]
Hm, can you post a photo of your TV's Settings > General > About this TV and Settings > Network ?

**Comment by peaoo** [comment]
![20160228_184304](https://cloud.githubusercontent.com/assets/17235241/13380810/60bfbbfe-de4b-11e5-8955-74899c59af45.jpg)

And it's software version 05.00.70

**Comment by WeeJeWel** [comment]
I've updated the App, and finding a TV should work now. There is a catch though:

I am not sure if sending commands works. It seems that those TV's have an older firmware version. Please try some commands and report back.

It would *really* help if you'd run the app from the command line, and paste the following lines when they appear:

```
connectToDevice { address: '192.168.0.109',
  UDN: 'uuid:a6104666-8bba-e593-bd99-2fd92bddc18c',
  modelNumber: 'WEBOS',
  modelURL: 'http://www.lge.com',
  modelName: 'LG Smart TV',
  modelDescription: '',
  manufacturerURL: 'http://www.lge.com',
  manufacturer: 'LG Electronics',
  friendlyName: 'emile-tv',
  deviceType: 'urn:dial-multiscreen-org:service:dial:1' }
```

**Comment by laurenssss** [comment]
not working for Lg 55lb670v also
web os 1.4.0-2515

**Comment by michelbliek** [comment]
Not working for  LG 49UF695V
Web OS 1.4.0-2507 (AFRO-ASHLEY)

"There were no new devices found! You may close this window"

**Comment by MarkWittens** [comment]
The app is not working on my LG 55UH615V as well, same problem as described here.

**Comment by Floodnl** [comment]
for the offline issue i found a sollution (atleast for my TV: LG oled65b6v) 
This occurs when the TV is turned off, and i do not understand why the following option works but: 

- Go to the settings menu of the TV (all settings) --> Go to 'Network' --> Go to 'LG Connect App' --> Turn it on. 

I was checking the app out on the TV and noticed i didn't had this option on. 

Turning this option on, the TV stayed paired for 2 weeks and ongoing now for me.
And i have no idea why this option would do so but it's the only option i'd tinkered with that has remotely something to do with the network. And the phone has been rebooted it might be so that the app perhaps keeps the connection open?

i hope this also works for you but ofcourse i cannot guarantee anything. 
Below i found a way to turn the TV on they also have a leaving connection open option in their server added for the TV if my workaround doesn't solve it this might: https://github.com/Danovadia/lgtv-http-server/commit/f4bbe58a8ade15a5882e16affe6cde5446a2c6d6

(full project in the box below)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------

an 'On' option is found by an community to turn the TV on

Description: After installing run `npm start` and check out `http://localhost:3000` for examples, or check the docs for HTTP requests. Can easily serve as a service for any kind of home automation (host on a Rasberry Pi).

source: https://github.com/Danovadia/lgtv-http-server

i have this running and it works like a charm using the website, i don't think it's much different then the homey app it's even on github.


PS offtopic  i haven't visited the forum on PC for a while (mobile0 but wow GJ guys looks way better.

**Comment by WeeJeWel** [comment]
Is anyone still experiencing problems with the latest version? https://apps.athom.com/app/com.lg.webos

**Comment by Floodnl** [comment]
Hi,  i am able to add the tv again, but after a couple of days the link between homey and the tv is gone again and we have to delete the tv and add it again. And this is repeating over and over again. This is one of the reasons I was a bit done with homey haven't do anything with it for a while

I also noticed the app no longer offers the possibility to mute / unmute the TV. So in have to play around with the volume to get this working again. 

Worst of all due to this app i choose an LG tv so it would work fine with homey and now it doesn't :) it's a bit of a bummer.

**Comment by WeeJeWel** [comment]
Thank you for your heads-up! I wasn't aware that the TV loses connection after a while. I will definitely investigate.

**Comment by Floodnl** [comment]
Btw Emile,  the volume adjustment is also not working for me. So since the mute option vanished i have no possibility to mute the tv while giving an command to homey.

And to complete te info:
oled65b6v  is my tv and it runs webos 3 do notice as many others are stating on the app store that the disconnect after a few days is nothing new the longest i kept it connected was about 2 weeks thinking i found a solution but it wasn't..

**Comment by WeeJeWel** [comment]
This should be fixed in v2.0.2, now available as beta at https://apps.athom.com/app/com.lg.webos

**Comment by Floodnl** [comment]
Dumb question maybe,  is this beta build only for the people running the beta homey build? 

The reason for this is that i am on the stable build and the v2.0.2 version crashes

**Comment by WeeJeWel** [comment]
My bad, fixed in v2.0.3 (app). No Homey beta is needed
